gdal_subsetter/transform.py

```diff
@@ -149,28 +146,25 @@ def process_item(self, item, source):
                          if 'data' in (v.roles or []))
 
             if self.config.fallback_authn_enabled:
-                input_filename = download(
-                    asset.href,
-                    output_dir,
-                    logger=self.logger,
-                    access_token=None,
-                    cfg=self.config)
-                input_filename = self.rename_file(input_filename, asset.href)
-                                
+                access_token = None
             else:
-                input_filename = download(
-                    asset.href,
-                    output_dir,
-                    logger=self.logger,
-                    access_token=self.message.accessToken,
-                    cfg=self.config)
-                input_filename = self.rename_file(input_filename, asset.href)
```

Comment:
The only difference between the code in these two conditions is the value used for the access token. Because of that, I updated the branches so that they only set the requisite access token, and then the download code is written only once.

(I do wonder if we can assume now that we will receive an access token all the time... something to ask the Harmony team)